src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs

```diff
@@ -2104,5 +2104,46 @@ class Point
             }
             Assert.Equal(expected.Length, i);
         }
+
+        [Fact, WorkItem(26613, "https://github.com/dotnet/roslyn/issues/26613")]
+        public void MissingDeconstruct_01()
+        {
+            var source =
+@"using System;
+public class C {
+    public void M() {
+        _ = this is (a: 1);
+    }
+}
+";
+            var compilation = CreateCompilation(source, options: TestOptions.ReleaseDll);
+            compilation.VerifyDiagnostics(
+                // (4,21): error CS8407: A single-element deconstruct pattern requires a type before the open parenthesis.
+                //         _ = this is (a: 1);
+                Diagnostic(ErrorCode.ERR_SingleElementPositionalPatternRequiresType, "(a: 1)").WithLocation(4, 21)
+                );
+        }
+
+        [Fact, WorkItem(26613, "https://github.com/dotnet/roslyn/issues/26613")]
+        public void MissingDeconstruct_02()
+        {
+            var source =
+@"using System;
+public class C {
+    public void M() {
+        _ = this is C(a: 1);
```

Comment:
Not related to this PR: the formatting that's currently checked in adds a space after the type: `_ = this is C (a: 1);`.
We can discuss if you think it should be adjusted.